tests/util/caches/test_descriptors.py

```diff
@@ -188,44 +199,47 @@ def fn(self, arg1, kwarg1=2):
         self.assertEqual(r, "fish")
         obj.mock.assert_not_called()
 
-    def test_cache_with_sync_exception(self):
+    def test_cache_with_sync_exception(self) -> None:
         """If the wrapped function throws synchronously, things should continue to work"""
 
         class Cls:
             @cached()
-            def fn(self, arg1):
+            def fn(self, arg1: int) -> None:
                 raise SynapseError(100, "mai spoon iz too big!!1")
 
         obj = Cls()
 
         # this should fail immediately
-        d = obj.fn(1)
+        # type-ignore: mypy thinks the RHS is None; we need to fix the annotation for
+        # @cached here.
```

Comment:
This sounds like the mypy plugin for cached isn't working *at all* then? This makes me very nervous to ignore TBH.

Reply:
I _think_ it works okay in Synapse proper, when `@cached` decorates an `async def` and is `await`ed. I don't think it works well if you're trying to get at the underlying `Deferred` machinery, as we are here.

Reply:
Does it Just Work™️ if we use `async def` instead?